Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -2426,4 +2426,13 @@ void WebsiteDataStore::setProxyConfigData(const API::Data& data, uuid_t proxyIde
 }
 #endif // HAVE(NW_PROXY_CONFIG)
 
+void WebsiteDataStore::setCompletionHandlerForRemovalFromNetworkProcess(CompletionHandler<void(String&&)>&& completionHandler)
+{
+    auto* networkProcess = networkProcessIfExists();
+    if (!networkProcess)
+        return completionHandler("Network process is not running"_s);
```

Comment:
Doesn't seem like something we ever want to happen. The client is not in charge on the network process. Odds are we should either launch the network process or we should be able to set the handler without launching the network process. This should not be an operation that can fail.

Reply:
My 2 cents. We can probably store the completion handler on the WebsiteDataStore and have `NetworkProcess::removeSession()` "take" it from the WebsiteDataStore. This would avoid having yet another HashMap on the NetworkProcessProxy.